stdlib/public/stubs/CommandLine.cpp

```diff
@@ -122,24 +122,30 @@ char ** _swift_stdlib_getUnsafeArgvArgc(int *outArgLen) {
                                            szArgList[i], -1, nullptr, 0,
                                            nullptr, nullptr);
     if (szBufferSize == 0) {
-      for (char *arg : argv)
+      for (char *arg : argv) {
         free(arg);
-      return nullptr;
+      }
+
+      break;
     }
 
     char *buffer = static_cast<char *>(calloc(static_cast<size_t>(szBufferSize),
                                               sizeof(char)));
     if (buffer == nullptr) {
-      for (char *arg : argv)
+      for (char *arg : argv) {
         free(arg);
-      return nullptr;
+      }
```

Comment:
This is an out of memory condition.  What is the behaviour on Darwin if there is an out of memory condition on launch?

Reply:
Various nastiness happens if we ever get `ENOMEM` back anywhere along these code paths on Darwin - data corruption, `std::bad_alloc` ripping through frames without unwinding info, etc. It seems like we should fatalError here.